src/main/java/ru/sbt/mipt/oop/SmartHomeDispatcher.java

```diff
@@ -0,0 +1,81 @@
+package ru.sbt.mipt.oop;
+import ru.sbt.mipt.oop.actions.*;
+import ru.sbt.mipt.oop.commands.CommandType;
+import ru.sbt.mipt.oop.commands.SensorCommand;
+import ru.sbt.mipt.oop.devices.Door;
+import ru.sbt.mipt.oop.devices.Light;
+import ru.sbt.mipt.oop.devices.Room;
+import ru.sbt.mipt.oop.devices.SmartHome;
+import ru.sbt.mipt.oop.events.SensorEvent;
+import ru.sbt.mipt.oop.events.SensorEventType;
+
+public class SmartHomeDispatcher {
+    private final SmartHome smartHome;
+
+    public SmartHomeDispatcher(SmartHome smartHome) {
+        this.smartHome = smartHome;
+    }
+
+    public SmartHome getSmartHome() {
+        return smartHome;
+    }
+
+    public void start() {
+        SensorEvent event = getNextSensorEvent();
+        while (event != null) {
+            handleEvent(event);
+            event = getNextSensorEvent();
+        }
+    }
+
+    private void handleEvent(SensorEvent event) {
+        System.out.println("Got event: " + event);
+        Action action;
+        switch (event.getType()) {
+            case LIGHT_ON:
+                 action = new LightOnAction(smartHome, event);
+                action.execute();
+                break;
+            case LIGHT_OFF:
+                action = new LightOffAction(smartHome, event);
+                action.execute();
+                break;
+            case DOOR_OPEN:
+                action = new DoorOpenedAction(smartHome, event);
+                action.execute();
+                break;
+            case DOOR_CLOSED:
+                action = new DoorClosedAction(smartHome, event);
+                action.execute();
+                for (Room room : smartHome.getRooms()) {
+                    for (Door door : room.getDoors()) {
+                        if (door.getId().equals(event.getObjectId()) && room.getName().equals("hall")) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0